server/mergin/sync/storages/disk.py

```diff
@@ -124,6 +132,26 @@ def _logger_callback(level, text_bytes):
 
         self.geodiff.set_logger_callback(_logger_callback)
 
+    @contextmanager
+    def geodiff_copy(self, file):
+        """Copy project file from live storage to geodiff temp storage for further actions.
+        If file cannot be copied then as a fallback it creates _tmp file next to original file.
+        Temporary copy is removed on exit.
+        """
+        file_path = file.replace(self.project_dir, "").lstrip(os.path.sep)
+        file_copy = os.path.join(self.geodiff_working_dir, file_path)
+        try:
+            copy_file(file, file_copy)
+            yield file_copy
+        except OSError as e:
+            logging.warning(f"Copy to geodiff dir failed: {str(e)}")
+            # fallback to live directory
+            file_copy = file + "_tmp"
```

Comment:
Why we need this temp copy?

Reply:
this will be copy of _any_ original file passed to geodiff (yes, for some cases we might not need it - this is for the sake of consistency though)